net.sf.eclipsecs.core/src/net/sf/eclipsecs/core/config/meta/MetadataFactory.java

```diff
@@ -551,12 +561,9 @@ private static void doInitialization() throws CheckstylePluginException {
    * @return
    */
   private static String groupId(String metadataFile) {
-    var start = metadataFile.indexOf("checks/") + 7;
-    var end = metadataFile.lastIndexOf("/");
-    if (start >= end) {
-      return "";
-    }
-    return metadataFile.substring(start, end);
+    String res = StringUtils.substringBetween(metadataFile, "/checks/", "/");
```

Comment:
for reviewers: the change in this method is not necessary for the fix, but it helps for debugging. the old code would produce groupIds like "ppycrawl.something..." because it just deleted 6 characters at the beginning when the indexOf returned -1. Now the groupId remains readable in that case, even though those confusng groupids are not used anywhere (the "normal" groupids like "annotations" remain as is).